The limitation was removed by https://github.com/tensorflow/tensorflow/commit/19b7519c2ec7b208aa6c1145132c84ebcbd795b0
The issue was likely "wrapt", not typing_extensions
See https://github.com/tensorflow/tensorflow/pull/61387 for wrapt
and https://github.com/tensorflow/tensorflow/issues/60687 for typing_extensions

Author: Alexander Grund (TU Dresden)

diff --git a/tensorflow/tools/pip_package/setup.py b/tensorflow/tools/pip_package/setup.py
index e77e6f81449..7927c8ac995 100644
--- a/tensorflow/tools/pip_package/setup.py
+++ b/tensorflow/tools/pip_package/setup.py
@@ -97,7 +97,7 @@ REQUIRED_PACKAGES = [
     'setuptools',
     'six >= 1.12.0',
     'termcolor >= 1.1.0',
-    'typing_extensions>=3.6.6,<4.6.0',
+    'typing_extensions >= 3.6.6',
     'wrapt >= 1.11.0',
     'tensorflow-io-gcs-filesystem >= 0.23.1;platform_machine!="arm64" or ' +
     'platform_system!="Darwin"',
